server/src/main/java/io/envoyproxy/controlplane/server/DiscoveryServer.java

```diff
@@ -207,7 +207,10 @@ private String send(Response response, String typeUrl) {
 
         LOGGER.info("[{}] response {} with nonce {} version {}", streamId, typeUrl, nonce, response.version());
 
-        responseObserver.onNext(discoveryResponse);
+        //Multiple updates might come - so need synchronized here
```

Comment:
This comment could be more descriptive, maybe something like this?

> The watch value streams are being observed on multiple threads, so we need to synchronize here because StreamObserver instances are not thread-safe.